src/flint/fmpz_factor.jl

```diff
@@ -17,20 +17,39 @@ function _factor(a::ZZRingElem)
   return res, canonical_unit(a)
 end
 
+# Just to give a helpful error message if someone tries to factor a boolean
+function factor(b::Bool)
+  throw(DomainError("Cannot factorize a boolean"));
+end
+
+# This function handles machine integer types (up to 64 bits)
 function factor(a::T) where T <: Union{Int, UInt}
   iszero(a) && throw(ArgumentError("Argument must be non-zero"))
   u = sign(a)
   a = u < 0 ? -a : a
   F = n_factor()
   @ccall libflint.n_factor(F::Ref{n_factor}, a::UInt)::Nothing
-  res = Dict{T, Int}()
+  res = Dict{T, Int}()  # factor-multiplicity pairs
   for i in 1:F.num
     z = F.p[i]
     res[z] = F.exp[i]
   end
   return Fac(u, res)
 end
 
+# This is supposed to be called only for T in [Int128, UInt128, BigInt]
+function factor(a::T) where T <: Integer
+  iszero(a) && throw(ArgumentError("Argument must be non-zero"))
+  u = sign(a)
+  F = factor(ZZ(abs(a)))
```

Comment:
Another thing you could try in a follow-up PR: if `T` is an Integer type that is "smaller" than `Int` or `UInt`, this might be more efficient:
```suggestion
  F = factor(flintify(abs(a)))
```
Check out the docstring for `Nemo.flintify` to get an explanation.